stock_picking_location_check/tests/test_stock_picking_location_check.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2024 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo.exceptions import UserError
+from odoo.tests.common import TransactionCase
+
+
+class TestStockPickingLocationCheck(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.product = cls.env["product.product"].create({"name": "Test Product"})
+        cls.stock_location = cls.env.ref("stock.stock_location_stock")
+        cls.shelf1_location = cls.env.ref("stock.stock_location_components")
+        cls.shelf2_location = cls.env.ref("stock.stock_location_14")
+        cls.refrigerator_small_location = cls.env.ref(
+            "stock.location_refrigerator_small"
+        )
```

Comment:
We should avoid using the demo data: https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#avoid-relying-on-demo-data

The current test setup makes the assumption obscure (how can one tell that `cls.refrigerator_small_location` is a child of `cls.shelf2_location` without looking at the demo data definition?) and therefore is not desirable.